onnx/backend/test/runner/__init__.py

```diff
@@ -155,6 +155,36 @@ def _assert_similar_outputs(ref_outputs, outputs):  # type: (Sequence[Any], Sequ
                 rtol=1e-3,
                 atol=1e-7)
 
+    def _retry_excute(self, func, times=3, **kwargs):  # type: (Callable[..., Any], int, **Any) -> Any
+        for i in range(times):
+            try:
+                return func(**kwargs)
+            except:
+                print('{} times tried'.format(i))
+                if i == times - 1:
+                    raise
+
+    def _download_model(self, model_test, model_dir, models_dir):  # type: (TestCase, Text, Text) -> None
+        try:
+            # On Windows, NamedTemporaryFile can not be opened for a
+            # second time
+            download_file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
in case `NamedTemoraryFile` throws, download_file will be not yet defined, and so in your finally cause, download_file will be an undefined variable.